### DISCUSSION TITLE
Add Cloudflare Argo as option to share site (interest check)

### DISCUSSION THREAD

**Discussion by nexxai** [laravel/valet#1141] [Ideas]
I prefer using Cloudflare Argo to share out my development sites since I get the benefit of using my own domain/subdomains while not having to pay a cent, and so I'm wondering if anyone else would be interested in me adding support for it to Valet.

I would use the ngrok implementation as a guide and make it work as similarly to it as possible.

**Comment by nexxai** [comment]
## Thoughts

Effectively, I would just be automating these steps:

1. brew install cloudflare/cloudflare/cloudflared
2. cloudflared tunnel login
3. cloudflared tunnel create examplesite
4. modify ~/.cloudflared/config.yml
5. cloudflared tunnel route dns examplesite examplesite.example.com
6. valet link examplesite.example.com
7. cloudflared tunnel run examplesite

**Comment by irazasyed** [comment]
Any update on this?

**Reply by irazasyed** [reply]
I modified the valet share command and created one for cloudflare. Currently using it in my projects. Script and instructions can be found here: https://gist.github.com/irazasyed/46295bc3b823bb3b13c9acd51b96b4ad

But obviously, if valet supports this officially, then it would be much better, and the process to install/setup is automated.
